install/src/Cabal.hs

```diff
@@ -37,18 +36,27 @@ cabalBuildHie versionNumber = do
       error (ghcVersionNotFoundFailMsg versionNumber)
     Just p -> return p
   execCabal_
-    ["new-build", "-w", ghcPath, "--write-ghc-environment-files=never", "--max-backjumps=5000"]
+    ["v2-build", "-w", ghcPath, "--write-ghc-environment-files=never", "--max-backjumps=5000", "--disable-tests"]
 
 cabalInstallHie :: VersionNumber -> Action ()
 cabalInstallHie versionNumber = do
+
   localBin <- getLocalBin
-  execCabal_
-    [ "new-install"
+  cabalVersion <- getCabalVersion
+  
+  let isCabal3 = checkVersion [3,0,0,0] cabalVersion
+      installDirOpt | isCabal3 = "--installdir"
+                    | otherwise = "--symlink-bindir"
+      installMethod | isWindowsSystem && isCabal3 = ["--install-method=copy"]
+                    | otherwise = []
```

Comment:
maybe more explicit `["--install-method=symlink"]`?

Reply:
Mmm, i think we shouldn't force users of other os's use symlinking cause maybe some of them will prefer copy them instead (modifying the cabal global config)